extension/src/lib/portBinding.js

```diff
@@ -13,51 +13,270 @@
 /* global fluid, chrome */
 "use strict";
 
-(function ($, fluid) {
+(function (fluid) {
     var gpii = fluid.registerNamespace("gpii");
 
-    // The main component to handle settings that require DOM manipulations.
-    // It contains various subcomponents for handling various settings.
+    /*
+        The `gpii.chrome.portBinding` grade provides a system for creating a port connection and sending/receiving
+        messages across that port. To verify that messages sent across a port are received and acted upon correctly,
+        posts provide a promise that is resolved/rejected based on a returned receipt. Connections are also set to
+        return receipts after an incoming message has been handled.
+
+        Workflow:
+        post - ( sends payload with message id )
+             - message id gets stored in a map with the promise to resolve/reject
+             -  receiver takes action of the message and returns a receipt ( includes the same message id )
+
+        receive - (receives a message)
+                - if it is a returned receipt
+                  - look into message map and resolve/reject corresponding promise
+                  - remove from map
+                - if it is a new message
+                  - take action on the message
+                  - send a return message containing the same message id
+
+
+        Post Messages:
+        1) Standard Message
+           - wait for receipt
+        2) Receipt
+           - do not wait for receipt
+
+        Received Messages:
+        1) Standard Message
+           - action, return receipt
+        2) Receipt
+           - remove from message map
+
+        To prevent handling receipts and messages that are broadcast but not intended for a connection, a whitelist
+        must be provided to allow `types` of requests to be handled.
+    */
+
     fluid.defaults("gpii.chrome.portBinding", {
         gradeNames: ["fluid.component"],
         connectionName: "",
+        messageType: "",
         members: {
-            lastIncomingMessage: {}
+            port: {
+                expander: {
+                    func: "{that}.setPort"
+                }
+            },
+            openRequests: {}
+        },
+        filters: {
+            messages: [],
```

Comment:
I think it is reasonable for the component to come with some out of the box defaults for messageType, messages, receipts, etc. together with good documentation about how all of these interact